framework/audio/internal/midiplayer.cpp

```diff
@@ -98,33 +103,31 @@ void Sequencer::requestData(tick_t tick)
 
     m_streamState.requested = true;
     m_midiStream->request.send(tick);
-    LOGD() << "request tick: " << tick;
 }
 
-void Sequencer::onChunkReceived(const Chunk& chunk)
+void MIDIPlayer::onChunkReceived(const Chunk& chunk)
 {
-    LOGD() << "chunk beginTick: " << chunk.beginTick;
     std::lock_guard<std::mutex> lock(m_dataMutex);
     m_midiData.chunks.insert({ chunk.beginTick, chunk });
     m_streamState.requested = false;
 }
 
-void Sequencer::process(float sec, Context* ctx)
+void MIDIPlayer::forwardTime(unsigned long miliseconds)
 {
-    if (m_status != Running) {
+    if (!isRunning()) {

```

Comment:
This is not thread-safe code.
Here the status is read from the audio worker thread.
But it changes in the main thread (from PlaybackController)

Reply:
fixed